CI: Test step for Bride sporadically fails

The tests for Bridge fail sporadically when running the CI pipeline on GH:
![image](https://user-images.githubusercontent.com/72415058/150482956-a8aa1bd6-13b7-488e-a416-063e0a2b000d.png)


```
Run docker run --rm -v "$PWD/shared:/shared" $IMAGE_NAME
yarn run v1.22.10
$ start-server-and-test 'yarn start:ci' http://localhost:3000 'yarn cypress:run'
1: starting server using command "yarn start:ci"
and when url "[ 'http://localhost:3000' ]" is responding with HTTP status code 200
running tests using command "yarn cypress:run"

$ ng serve --port=3000 --no-live-reload --configuration=test
Compiling @angular/core : es2015 as esm2015
Compiling @angular/common : es2015 as esm2015
Compiling @angular/platform-browser : es2015 as esm2015
Compiling @angular/platform-browser-dynamic : es2015 as esm2015
Compiling @angular/flex-layout/core : es2015 as esm2015
Compiling @angular/flex-layout/extended : es2015 as esm2015
Compiling @angular/cdk/bidi : es2015 as esm2015
Compiling @angular/flex-layout/flex : es2015 as esm2015
Compiling @angular/flex-layout/grid : es2015 as esm2015
Compiling @angular/flex-layout : es2015 as esm2015
Compiling @angular/cdk/platform : es2015 as esm2015
Compiling @angular/cdk/collections : es2015 as esm2015
Compiling @angular/cdk/scrolling : es2015 as esm2015
Compiling @angular/cdk/portal : es2015 as esm2015
Compiling @angular/cdk/keycodes : es2015 as esm2015
Compiling @angular/cdk/overlay : es2015 as esm2015
Compiling @angular/cdk/observers : es2015 as esm2015
Compiling @angular/cdk/a11y : es2015 as esm2015
Compiling @angular/animations : es2015 as esm2015
Compiling @angular/animations/browser : es2015 as esm2015
Compiling @angular/platform-browser/animations : es2015 as esm2015
Compiling @angular/forms : es2015 as esm2015
Compiling @angular/material/core : es2015 as esm2015
Compiling @angular/material/dialog : es2015 as esm2015
Compiling @dynatrace/barista-components/card : es2015 as esm2015
Compiling @angular/cdk/tree : es2015 as esm2015
Compiling @dynatrace/barista-components/core : es2015 as esm2015
Compiling @angular/common/http : es2015 as esm2015
Compiling @dynatrace/barista-components/icon : es2015 as esm2015
Compiling @dynatrace/barista-components/button : es2015 as esm2015
Compiling @dynatrace/barista-components/formatters : es2015 as esm2015
Compiling @dynatrace/barista-components/loading-distractor : es2015 as esm2015
Compiling @dynatrace/barista-components/theming : es2015 as esm2015
Compiling @dynatrace/barista-components/chart : es2015 as esm2015
Compiling @dynatrace/barista-components/confirmation-dialog : es2015 as esm2015
Compiling @dynatrace/barista-components/progress-bar : es2015 as esm2015
Compiling @dynatrace/barista-components/consumption : es2015 as esm2015
Compiling @dynatrace/barista-components/context-dialog : es2015 as esm2015
Compiling @dynatrace/barista-components/expandable-text : es2015 as esm2015
Compiling @dynatrace/barista-components/expandable-panel : es2015 as esm2015
Compiling @dynatrace/barista-components/expandable-section : es2015 as esm2015
Compiling @dynatrace/barista-components/info-group : es2015 as esm2015
Compiling @dynatrace/barista-components/key-value-list : es2015 as esm2015
Compiling @dynatrace/barista-components/menu : es2015 as esm2015
Compiling @dynatrace/barista-components/progress-circle : es2015 as esm2015
Compiling @dynatrace/barista-components/form-field : es2015 as esm2015
Compiling @dynatrace/barista-components/select : es2015 as esm2015
Compiling @dynatrace/barista-components/show-more : es2015 as esm2015
Compiling @dynatrace/barista-components/switch : es2015 as esm2015
Compiling @angular/cdk/text-field : es2015 as esm2015
Compiling @dynatrace/barista-components/input : es2015 as esm2015
Compiling @dynatrace/barista-components/tag : es2015 as esm2015
Compiling @dynatrace/barista-components/top-bar-navigation : es2015 as esm2015
Compiling @angular/cdk/clipboard : es2015 as esm2015
Compiling @dynatrace/barista-components/copy-to-clipboard : es2015 as esm2015
Compiling @dynatrace/barista-components/toggle-button-group : es2015 as esm2015
Compiling @dynatrace/barista-components/autocomplete : es2015 as esm2015
Compiling @dynatrace/barista-components/button-group : es2015 as esm2015
Compiling @dynatrace/barista-components/checkbox : es2015 as esm2015
Compiling @dynatrace/barista-components/overlay : es2015 as esm2015
Compiling @dynatrace/barista-components/highlight : es2015 as esm2015
Compiling @dynatrace/barista-components/filter-field : es2015 as esm2015
Compiling @angular/cdk/layout : es2015 as esm2015
Compiling @dynatrace/barista-components/drawer : es2015 as esm2015
Compiling @dynatrace/barista-components/radio : es2015 as esm2015
Compiling @dynatrace/barista-components/quick-filter : es2015 as esm2015
Compiling @dynatrace/barista-components/tile : es2015 as esm2015
Compiling @dynatrace/barista-components/toast : es2015 as esm2015
Compiling ngx-moment : es2015 as esm2015
Compiling @angular/router : es2015 as esm2015
Compiling @dynatrace/barista-components/alert : es2015 as esm2015
Compiling @angular/cdk/table : es2015 as esm2015
Compiling @angular/cdk/drag-drop : es2015 as esm2015
Compiling @dynatrace/barista-components/indicator : es2015 as esm2015
Compiling @dynatrace/barista-components/empty-state : es2015 as esm2015
Compiling @dynatrace/barista-components/pagination : es2015 as esm2015
Compiling @dynatrace/barista-components/table : es2015 as esm2015
Compiling @dynatrace/barista-components/tree-table : es2015 as esm2015
- Generating browser application bundles...
✔ Browser application bundle generation complete.

Initial Chunk Files | Names         |      Size
vendor.js           | vendor        |  10.93 MB
main.js             | main          |   1.94 MB
polyfills.js        | polyfills     | 137.08 kB
styles.css          | styles        | 119.22 kB
runtime.js          | runtime       |   6.15 kB

| Initial Total |  13.13 MB

Build at: 2022-01-21T06:39:36.684Z - Hash: bafb3fcd50d0123ccccd - Time: 54376ms

Warning: /usr/src/app/client/app/not-found/not-found.component.ts depends on 'rxjs/internal/Subject'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies



** Angular Live Development Server is listening on localhost:3000, open your browser on http://localhost:3000/ **


✔ Compiled successfully.
$ cypress run --browser=chrome
[STARTED] Task without title.
- Generating browser application bundles...
[TITLE]  Verified Cypress!       /root/.cache/Cypress/9.2.1/Cypress
[SUCCESS]  Verified Cypress!       /root/.cache/Cypress/9.2.1/Cypress
[266:0121/063944.547869:ERROR:bus.cc(392)] Failed to connect to the bus: Address does not contain a colon
[266:0121/063944.547951:ERROR:bus.cc(392)] Failed to connect to the bus: Address does not contain a colon
[266:0121/063944.548482:ERROR:bus.cc(392)] Failed to connect to the bus: Failed to connect to socket /var/run/dbus/system_bus_socket: No such file or directory
[428:0121/063944.591221:ERROR:gpu_init.cc(453)] Passthrough is not supported, GL is swiftshader, ANGLE is 
✔ Browser application bundle generation complete.

Initial Chunk Files | Names  |      Size
styles.css          | styles | 119.22 kB

4 unchanged chunks

Build at: 2022-01-21T06:39:45.580Z - Hash: 563ac690ccb9ee99b16c - Time: 7038ms

✔ Compiled successfully.

================================================================================

  (Run Starting)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Cypress:        9.2.1                                                                          │
  │ Browser:        Chrome 91 (headless)                                                           │
  │ Node Version:   v14.17.0 (/usr/local/bin/node)                                                 │
  │ Specs:          10 found (app.spec.ts, git-credentials-edit.spec.ts, navigation-buttons-evalua │
  │                 tion-screen-test.ts, navigation-project-change.spec.ts, project-create.spec.ts │
  │                 , project-delete.spec.ts, secret-add-delete.spec.ts, sequences.spec.ts, unifor │
  │                 m-integrations.spec.ts,...)                                                    │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


────────────────────────────────────────────────────────────────────────────────────────────────────
                                                                                                    
  Running:  app.spec.ts                                                                    (1 of 10)


  Loading Bridge Example
    ✓ should match the title (2465ms)


  1 passing (3s)


  (Results)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Tests:        1                                                                                │
  │ Passing:      1                                                                                │
  │ Failing:      0                                                                                │
  │ Pending:      0                                                                                │
  │ Skipped:      0                                                                                │
  │ Screenshots:  1                                                                                │
  │ Video:        false                                                                            │
  │ Duration:     2 seconds                                                                        │
  │ Spec Ran:     app.spec.ts                                                                      │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


  (Screenshots)

  -  /usr/src/app/dist/cypress/screenshots/app.spec.ts/entry-page.png                     (1280x720)


────────────────────────────────────────────────────────────────────────────────────────────────────
                                                                                                    
  Running:  git-credentials-edit.spec.ts                                                   (2 of 10)


  Changing git credentials
    ✓ The test changes git credentials and makes sure they changed successfully (4063ms)
    1) "after each" hook for "Prevent data loss if git crendetials not saved before navigation"


  1 passing (7s)
  1 failing

  1) Changing git credentials
       "after each" hook for "Prevent data loss if git crendetials not saved before navigation":
     AssertionError: expected error to have been called exactly "0 times", but it was called once

    The following calls were made:

    error("ERROR", Object{8}) at console.error (http://localhost:3000/__cypress/runner/cypress_runner.js:140867:26)

Because this error occurred during a `after each` hook we are skipping all of the remaining tests.
      at Context.eval (http://localhost:3000/__cypress/tests?p=cypress/support/index.ts:131:46)




  (Results)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Tests:        2                                                                                │
  │ Passing:      1                                                                                │
  │ Failing:      1                                                                                │
  │ Pending:      0                                                                                │
  │ Skipped:      0                                                                                │
  │ Screenshots:  1                                                                                │
  │ Video:        false                                                                            │
  │ Duration:     7 seconds                                                                        │
  │ Spec Ran:     git-credentials-edit.spec.ts                                                     │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


  (Screenshots)

  -  /usr/src/app/dist/cypress/screenshots/git-credentials-edit.spec.ts/Prevent data      (1280x720)
     loss if git crendetials not saved before navigation -- after each hook (failed).               
     png                                                                                            


────────────────────────────────────────────────────────────────────────────────────────────────────
                                                                                                    
  Running:  navigation-buttons-evaluation-screen-test.ts                                   (3 of 10)


  Test Navigation Buttons In Evaluation Screen
    ✓ The test clicks on Navigation buttons and make sure the pages are aopen respectivily  (12178ms)


  1 passing (12s)


  (Results)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Tests:        1                                                                                │
  │ Passing:      1                                                                                │
  │ Failing:      0                                                                                │
  │ Pending:      0                                                                                │
  │ Skipped:      0                                                                                │
  │ Screenshots:  0                                                                                │
  │ Video:        false                                                                            │
  │ Duration:     12 seconds                                                                       │
  │ Spec Ran:     navigation-buttons-evaluation-screen-test.ts                                     │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


────────────────────────────────────────────────────────────────────────────────────────────────────
                                                                                                    
  Running:  navigation-project-change.spec.ts                                              (4 of 10)


  Test project navigation
    ✓ should correctly load service states of different service (3812ms)


  1 passing (4s)


  (Results)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Tests:        1                                                                                │
  │ Passing:      1                                                                                │
  │ Failing:      0                                                                                │
  │ Pending:      0                                                                                │
  │ Skipped:      0                                                                                │
  │ Screenshots:  0                                                                                │
  │ Video:        false                                                                            │
  │ Duration:     3 seconds                                                                        │
  │ Spec Ran:     navigation-project-change.spec.ts                                                │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


────────────────────────────────────────────────────────────────────────────────────────────────────
                                                                                                    
  Running:  project-create.spec.ts                                                         (5 of 10)


  Create new project test
    ✓ test new project create (4005ms)


  1 passing (4s)


  (Results)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Tests:        1                                                                                │
  │ Passing:      1                                                                                │
  │ Failing:      0                                                                                │
  │ Pending:      0                                                                                │
  │ Skipped:      0                                                                                │
  │ Screenshots:  0                                                                                │
  │ Video:        false                                                                            │
  │ Duration:     4 seconds                                                                        │
  │ Spec Ran:     project-create.spec.ts                                                           │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


────────────────────────────────────────────────────────────────────────────────────────────────────
                                                                                                    
  Running:  project-delete.spec.ts                                                         (6 of 10)


  Project delete test
    ✓ test (4001ms)


  1 passing (4s)


  (Results)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Tests:        1                                                                                │
  │ Passing:      1                                                                                │
  │ Failing:      0                                                                                │
  │ Pending:      0                                                                                │
  │ Skipped:      0                                                                                │
  │ Screenshots:  0                                                                                │
  │ Video:        false                                                                            │
  │ Duration:     4 seconds                                                                        │
  │ Spec Ran:     project-delete.spec.ts                                                           │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


────────────────────────────────────────────────────────────────────────────────────────────────────
                                                                                                    
  Running:  secret-add-delete.spec.ts                                                      (7 of 10)


  Keptn Secrets adding deleting test
    ✓ The test goes over the pages and does validation (6473ms)


  1 passing (7s)


  (Results)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Tests:        1                                                                                │
  │ Passing:      1                                                                                │
  │ Failing:      0                                                                                │
  │ Pending:      0                                                                                │
  │ Skipped:      0                                                                                │
  │ Screenshots:  0                                                                                │
  │ Video:        false                                                                            │
  │ Duration:     6 seconds                                                                        │
  │ Spec Ran:     secret-add-delete.spec.ts                                                        │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


────────────────────────────────────────────────────────────────────────────────────────────────────
                                                                                                    
  Running:  sequences.spec.ts                                                              (8 of 10)


  Sequences
    ✓ should show a loading indicator when sequences are not loaded (2148ms)
    ✓ should show an empty state if no sequences are loaded (1126ms)
    ✓ should show a list of sequences if everything is loaded (855ms)
    ✓ should show a filtered list if filters are applied (6055ms)


  4 passing (11s)


  (Results)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Tests:        4                                                                                │
  │ Passing:      4                                                                                │
  │ Failing:      0                                                                                │
  │ Pending:      0                                                                                │
  │ Skipped:      0                                                                                │
  │ Screenshots:  0                                                                                │
  │ Video:        false                                                                            │
  │ Duration:     10 seconds                                                                       │
  │ Spec Ran:     sequences.spec.ts                                                                │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


────────────────────────────────────────────────────────────────────────────────────────────────────
                                                                                                    
  Running:  uniform-integrations.spec.ts                                                   (9 of 10)


  Integrations
    ✓ should be on page uniform (1944ms)
    ✓ should select an integration and show related subscriptions (1434ms)
    ✓ should have disabled buttons for a subscription, when subscription id is not given (922ms)
    ✓ should add a simple subscription (1976ms)
    ✓ should add a webhook subscription (6093ms)
    ✓ should delete a subscription (1381ms)
    ✓ should edit a subscription (1840ms)


  7 passing (16s)


  (Results)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Tests:        7                                                                                │
  │ Passing:      7                                                                                │
  │ Failing:      0                                                                                │
  │ Pending:      0                                                                                │
  │ Skipped:      0                                                                                │
  │ Screenshots:  0                                                                                │
  │ Video:        false                                                                            │
  │ Duration:     16 seconds                                                                       │
  │ Spec Ran:     uniform-integrations.spec.ts                                                     │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


────────────────────────────────────────────────────────────────────────────────────────────────────
                                                                                                    
  Running:  user-menu.e2e-spec.ts                                                         (10 of 10)


  User Menu
    ✓ should open the user menu validate api token (2449ms)


  1 passing (3s)


  (Results)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Tests:        1                                                                                │
  │ Passing:      1                                                                                │
  │ Failing:      0                                                                                │
  │ Pending:      0                                                                                │
  │ Skipped:      0                                                                                │
  │ Screenshots:  1                                                                                │
  │ Video:        false                                                                            │
  │ Duration:     2 seconds                                                                        │
  │ Spec Ran:     user-menu.e2e-spec.ts                                                            │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


  (Screenshots)

  -  /usr/src/app/dist/cypress/screenshots/user-menu.e2e-spec.ts/user-menu-open-with-     (1280x720)
     api-command-revealed.png                                                                       


================================================================================

  (Run Finished)


       Spec                                              Tests  Passing  Failing  Pending  Skipped  
  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ ✔  app.spec.ts                              00:02        1        1        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✖  git-credentials-edit.spec.ts             00:07        2        1        1        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  navigation-buttons-evaluation-scree      00:12        1        1        -        -        - │
  │    n-test.ts                                                                                   │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  navigation-project-change.spec.ts        00:03        1        1        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  project-create.spec.ts                   00:04        1        1        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  project-delete.spec.ts                   00:04        1        1        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  secret-add-delete.spec.ts                00:06        1        1        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  sequences.spec.ts                        00:10        4        4        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  uniform-integrations.spec.ts             00:16        7        7        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  user-menu.e2e-spec.ts                    00:02        1        1        -        -        - │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
    ✖  1 of 10 failed (10%)                     01:10       20       19        1        -        -  

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Error: Command failed with exit code 1: yarn cypress:run
    at makeError (/usr/src/app/node_modules/execa/lib/error.js:60:11)
    at handlePromise (/usr/src/app/node_modules/execa/index.js:118:26)
    at processTicksAndRejections (internal/process/task_queues.js:95:5) {
  shortMessage: 'Command failed with exit code 1: yarn cypress:run',
  command: 'yarn cypress:run',
  escapedCommand: '"yarn cypress:run"',
  exitCode: 1,
  signal: undefined,
  signalDescription: undefined,
  stdout: undefined,
  stderr: undefined,
  failed: true,
  timedOut: false,
  isCanceled: false,
  killed: false
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Error: Process completed with exit code 1.
```

--- a/bridge/client/app/_components/ktb-project-settings-git/ktb-project-settings-git.component.html
+++ b/bridge/client/app/_components/ktb-project-settings-git/ktb-project-settings-git.component.html
@@ -1,9 +1,9 @@
 <h2>Git upstream repository</h2>
 <p class="mt-0 mb-3">
   It is recommended to set a Git upstream repository. In future releases, projects without a Git repository will not be
-  supported. Please specify the Git remote url, the user and the token. Instructions, how to set your Git provider can
+  supported. Please specify the Git remote url, the user and the token. Instructions on how to set your Git provider can
   be found in the
-  <a [href]="'/manage/git_upstream/' | keptnUrl" target="_blank">Git-based upstream documentation</a>
+  <a [href]="'/manage/git_upstream/' | keptnUrl" target="_blank">Git-based upstream documentation</a>.
 </p>
 <form
   [formGroup]="gitUpstreamForm"

--- a/bridge/client/app/_components/ktb-project-settings-git/ktb-project-settings-git.component.html
+++ b/bridge/client/app/_components/ktb-project-settings-git/ktb-project-settings-git.component.html
@@ -1,7 +1,8 @@
 <h2>Git upstream repository</h2>
 <p class="mt-0 mb-3">
-  It is recommended to set a Git upstream repository. Please specify the Git remote url, the user and the token.
-  <br />Instructions, how to set your Git provider can be found in the
+  It is recommended to set a Git upstream repository. In future releases, projects without a Git repository will not be
+  supported. Please specify the Git remote url, the user and the token. Instructions, how to set your Git provider can
+  be found in the
   <a [href]="'/manage/git_upstream/' | keptnUrl" target="_blank">Git-based upstream documentation</a>
 </p>
 <form

--- a/bridge/client/app/_components/ktb-project-tile/ktb-project-tile.component.html
+++ b/bridge/client/app/_components/ktb-project-tile/ktb-project-tile.component.html
@@ -44,14 +44,14 @@
         (click)="$event.stopPropagation()"
       ></a>
     </p>
-    <div class="mb-2" fxLayout="row" fxLayoutAlign="flex-start center" *ngIf="!project.gitRemoteURI">
-      <dt-icon class="info mr-1" [name]="'information'"></dt-icon>
+    <div class="mb-2" fxLayout="row" fxLayoutAlign="flex-start start" *ngIf="!project.gitRemoteURI">
+      <dt-icon class="event-icon warning mr-1" [name]="'incident'"></dt-icon>
       <p class="small m-0" uitestid="keptn-project-tile-gitRemoteUri-noUpstreamConfigured">
-        No Git upstream configured.
+        Keptn will not support a project without Git upstream repository in future releases.<br />
         <a
           [routerLink]="['/project', project.projectName, 'settings', 'project']"
           uitestid="keptn-project-tile-gitRemoteUri-setUpstream"
-          >Set Git upstream</a
+          >Set the Git upstream of your project</a
         >
       </p>
     </div>

--- a/bridge/cypress/integration/git-credentials-edit.spec.ts
+++ b/bridge/cypress/integration/git-credentials-edit.spec.ts
@@ -96,6 +96,8 @@ describe('Changing git credentials', () => {
       statusCode: 200,
     });
 
+    cy.intercept('GET', '/api/project/dynatrace/serviceStates', { statusCode: 200 });
+
     cy.visit('/');
     cy.wait('@metadataCmpl');
     basePage.selectProject(DYNATRACE_PROJECT);

